presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -144,6 +145,16 @@ default int compareTo(int leftPosition, int leftOffset, int leftLength, Block ri
     }
 
     /**
+     * {@code consumer} visits each of the internal data container and accepts the size for it.
+     * Returns the remaining retained size that has not been visited.
+     * This method can be helpful in cases such as memory counting for internal data structure.
+     */
+    default long getRetainedSizeInBytesBreakdown(BiConsumer<Object, Integer> consumer)
+    {
+        throw new UnsupportedOperationException(getClass().getName());
+    }
```

Comment:
@nezihyigitbasi has a suggestion changing the interface to `void` so that the function only gets the sizes for internal arrays, slices, blocks, etc. A new interface (maybe called `getInstanceSizeInBytes()`) can be created for obtaining the instance size of a block. @haozhun and I also feel the separation could make the interfaces cleaner. @dain, what do you think?